test/e2e/config.go

```
@@ -0,0 +1,129 @@
+	currentBasename := os.Getenv(ImageBasenameEnv)
+	if currentBasename != "" && currentBasename != productionBasename {
+		// IMAGE_BASENAME is set and differs from production, use it
+		return currentBasename, nil
+	}
+
+	// Try to detect local registry and construct basename
+	if isLocalRegistryAvailable() {
+		return DefaultLocalRegistry + "/knative-serving-wasm", nil
+	}
+
+	return "", fmt.Errorf(
```

golangci-lint (2.5.0)

[error] do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\n\t\"e2e tests require %s environment variable or local registry on %s; \"+\n\t\t\"alternatively, set %s to a non-production value; \"+\n\t\t\"production value is %q\",\n\tE2EImageBasenameEnv, DefaultLocalRegistry, ImageBasenameEnv, productionBasename,\n)"

(err113)